hibernation-setup-tool.c

```diff
@@ -1566,14 +1572,15 @@ static int handle_pre_systemd_suspend_notification(const char *action)
         close(fd);
 
         if (unlink(hibernate_lock_file_name) < 0) {
-            if (errno != EEXIST) {
+            if (!access(hibernate_lock_file_name, F_OK)) {
                 notify_vm_host(HOST_VM_NOTIFY_PRE_HIBERNATION_FAILED);
                 log_fatal("Couldn't remove %s: %s", hibernate_lock_file_name, strerror(errno));
             }
         }
-        if (link(pattern, hibernate_lock_file_name) < 0) {
```

Comment:
```
        // This throws an invalid-cross device link error, because we are creating a hard link 
        // of a file on /tmp partition, tmpfs filesystem to /etc partition (on ubuntu ext4 filesystem)
        // While hardlinks cannot work across filesystem boundaries, symbolic links can.
        // This is because hard link knows nothing about what is on the other filesystem
        // or where your data there is stored. 
        // link(pattern, hibernate_lock_file_name) -> symlink(pattern, hibernate_lock_file_name)
```